File: src/main/java/botmanager/frostbalance/commands/meta/HelpCommand.java
package botmanager.frostbalance.commands.meta;

import botmanager.frostbalance.Frostbalance;
import botmanager.frostbalance.command.AuthorityLevel;
import botmanager.frostbalance.command.FrostbalanceCommandBase;
import botmanager.frostbalance.command.FrostbalanceHybridCommandBase;
import botmanager.frostbalance.command.GenericMessageReceivedEventWrapper;

/**
 *
 * @author MC_2018 <mc2018.git@gmail.com>
 */
public class HelpCommand extends FrostbalanceHybridCommandBase {

    public HelpCommand(Frostbalance bot) {
        super(bot, new String[] {
                bot.getPrefix() + "help"
        }, AuthorityLevel.GENERIC);
    }

    @Override
    protected void runHybrid(GenericMessageReceivedEventWrapper eventWrapper, String... params) {
        final String[] result = {"__**Frostbalance**__\n\n"};

        for (FrostbalanceCommandBase command : bot.getCommands()) {
            command.getInfo(eventWrapper).ifPresent(info -> result[0] += info + "\n");
        }

        eventWrapper.sendResponse(result[0]);
    }

    @Override
    public String info(AuthorityLevel authorityLevel, boolean isPublic) {
        if (isPublic) {
            return "**" + bot.getPrefix() + "help** in public chat - rattles off a bunch of useless commands\n" +
                    "**" + bot.getPrefix() + "help** in DM - lists what commands work in PM";
        } else {
            return "**" + bot.getPrefix() + "help** in DM - rattles off a bunch of useless commands\n" +
                    "**" + bot.getPrefix() + "help** in public chat - lists what commands work in public chat";
        }
    }
}

File: src/main/java/botmanager/frostbalance/commands/meta/HelpCommand.kt
package botmanager.frostbalance.commands.meta

import botmanager.frostbalance.Frostbalance
import botmanager.frostbalance.command.*

/**
 *
 * @author MC_2018 <mc2018.git></mc2018.git>@gmail.com>
 */
class HelpCommand(bot: Frostbalance) : FrostbalanceGuildCommand(bot, arrayOf(
        "help"
), AuthorityLevel.GENERIC, ContextLevel.ANY) {
    override fun executeWithGuild(context: GuildMessageContext, vararg params: String) {
        var arguments = ArgumentStream(params)

        arguments.next()?.let {
            bot.commands.firstOrNull { command -> command.allAliases.contains(it) }
                    ?.let {
                        context.sendResponse(it.getInfo(context))
                        return
                    }
            return context.sendResponse("Could not find command $it.")
        }

        var result: MutableList<String> = mutableListOf()
        for (command in bot.commands) {
            val info = command.getInfo(context)
            if (info != null) {
                result.add(info)
            }
        }
        context.sendMultiLineResponse(result)
    }

    public override fun info(authorityLevel: AuthorityLevel?, isPublic: Boolean): String? {
        return if (isPublic) {
            """
     **${bot.prefix}help** in public chat - rattles off a bunch of useless commands
     **${bot.prefix}help** in DM - lists what commands work in PM
     """.trimIndent()
        } else {
            """
     **${bot.prefix}help** in DM - rattles off a bunch of useless commands
     **${bot.prefix}help** in public chat - lists what commands work in public chat
     """.trimIndent()
        }
    }
}